### PR TITLE
Disable torchrec/sparse from top-level Dynamo tracing

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #105733

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

### CODE REVIEW

File: torch/_dynamo/skipfiles.py [hunk 1]
```diff
@@ -163,6 +163,12 @@ def _module_dir(m: types.ModuleType):
 FBCODE_SKIP_DIRS = {
     "torchrec/distributed",
     "torchrec/fb/distributed",
+    # JaggedTensor and KeyedJaggedTensor are fairly badly behaved, because
+    # they tend to contain extremely long lists of integers that vary from
+    # run to run.  We are interested in tracing models that contain references
+    # to KJTs, but tracing an individual KeyedJaggedTensor is rarely
```
Comment (anijain2305) [inline torch/_dynamo/skipfiles.py:169]: Wondering whats the story for export here.

File: torch/_dynamo/skipfiles.py [hunk 2]
```diff
@@ -163,6 +163,12 @@ def _module_dir(m: types.ModuleType):
...
+    # profitable
+    "torchrec/sparse",
```
Comment (voznesenskym) [inline torch/_dynamo/skipfiles.py:171]: Feels like it should be config driven, or easily modifiable in fbcode without changing dynamo source.

Comment (ezyang) [reply torch/_dynamo/skipfiles.py:171]: Look, if you want to put it in configerator, I'll put it in configerator, but you are still going to have to review my config modifications

File: torch/_dynamo/skipfiles.py (hunk 1 above)
Comment (ezyang) [reply torch/_dynamo/skipfiles.py:169]: for export you aren't going to start the export from jagged_tensor.py anyway, so this heuristic is... not relevant for this (legitimate) question

File: torch/_dynamo/skipfiles.py (hunk 2 above)
Comment (voznesenskym) [reply torch/_dynamo/skipfiles.py:171]: I meant put it in _dynamo.config

File: torch/_dynamo/skipfiles.py (hunk 1 above)
Comment (voznesenskym) [reply torch/_dynamo/skipfiles.py:169]: To clarify @anijain2305 - skipfiles are only consulted when we decide where to enter. The export entry point will almost certainly never be a skipfile, and so this should be sound for that.

@ezyang this *might* still cause issues if you are inlining in the middle of export. Inlining checks skipfiles.

**Review by voznesenskym — APPROVED** [review]
Fine because this never traced well anyway, and if export cares they can carve out.

File: torch/_dynamo/skipfiles.py (hunk 1 above)
Comment (ezyang) [reply torch/_dynamo/skipfiles.py:169]: Oh, that sounds bad, I don't want export to not work haha (not that it works today anyway, but...)

**Pull request closed without merging**